02-oop-inheritance/bundle.js

```diff
@@ -0,0 +1,200 @@
+
+var _get = function get(object, property, receiver) { if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { return get(parent, property, receiver); } } else if ("value" in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } };
+
+function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
+
+function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }
+
+function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }
+
+var _EventEmitter = require('./eventemitter.js');
+var EventEmitter = _EventEmitter.EventEmitter;
+
+var Movie = function (_EventEmitter2) {
+  _inherits(Movie, _EventEmitter2);
+
+  function Movie(title, year, duration) {
+    _classCallCheck(this, Movie);
+
+    var _this = _possibleConstructorReturn(this, (Movie.__proto__ || Object.getPrototypeOf(Movie)).call(this));
+
+    _this.title = title;
+    _this.year = year;
+    _this.duration = duration;
+    _this.cast = [];
+    return _this;
+  }
+
+  _createClass(Movie, [{
+    key: 'play',
+    value: function play() {
+      _get(Movie.prototype.__proto__ || Object.getPrototypeOf(Movie.prototype), 'emit', this).call(this, "play");
+    }
+  }, {
+    key: 'pause',
+    value: function pause() {
+      _get(Movie.prototype.__proto__ || Object.getPrototypeOf(Movie.prototype), 'emit', this).call(this, "pause");
+    }
+  }, {
+    key: 'resume',
+    value: function resume() {
+      _get(Movie.prototype.__proto__ || Object.getPrototypeOf(Movie.prototype), 'emit', this).call(this, "resume");
```

Comment:
here we have 3 cases where you call the same function with the same validation. and the only thing that change is the parameter in the `call` function. there's a better way to do that.